### PR TITLE
Revert "Query compilation performance (#430)"

### PR DESCRIPTION
**Pull request by smitpatel**
This reverts commit 248330348ea7b5d756fa4bab52d5cebc98887fde.

### CODE REVIEW

File: src/EntityFramework/Core/Query/PlanCompiler/PlanCompiler.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ internal class PlanCompiler
         // Determines the maximum size of the query in terms of Iqt nodes for which we attempt to do transformation rules.
         // This number is ignored if applyTransformationsRegardlessOfSize is enabled.
         // </summary>
-        private const int MaxNodeCountForTransformations = 10000;
+        private const int MaxNodeCountForTransformations = 100000;
```
Comment (smitpatel) [inline src/EntityFramework/Core/Query/PlanCompiler/PlanCompiler.cs]: This line @bricelam?

**Comment by smitpatel** [comment]
cc: @divega - This needs to go in 6.3 release.

File: src/EntityFramework/Core/Query/PlanCompiler/PlanCompiler.cs (hunk 1 above)
Comment (bricelam) [reply src/EntityFramework/Core/Query/PlanCompiler/PlanCompiler.cs]: Yep. Keep it at the lower value.

**Comment by Pilchie** [comment]
Can we get an ask mode template filled out for this?

**Comment by smitpatel** [comment]
@pilchie - This fixes #1217 a customer reported issue. Arthur used to fill out ask mode template for us, but in the absence we hoped that you could do so for us (else @divega is our backup).

**Comment by Pilchie** [comment]
# Ask mode template 

#### Description
Revert an optimization PR that caused regressions from shipping EF.
 
#### Customer Impact
Found by a few customers using EF6.
 
#### Regression?
Yes, regression from EF 6.2, introduced in June. 

#### Risk
Low – it’s a straight revert of the original PR to get back to shipped behaviour

**Comment by divega** [comment]
This approved.

**Pull request merged**